src/main/kotlin/org/opensearch/observability/validation/IntegrationValidator.kt

```diff
@@ -0,0 +1,56 @@
+package org.opensearch.observability.validation
+
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.core.JsonParseException
+import com.fasterxml.jackson.databind.JsonMappingException
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.worldturner.medeia.api.UrlSchemaSource
+import com.worldturner.medeia.api.ValidationFailedException
+import com.worldturner.medeia.api.jackson.MedeiaJacksonApi
+import com.worldturner.medeia.schema.validation.SchemaValidator
+import org.opensearch.commons.utils.logger
+
+class IntegrationValidator(val config: String) {
+    companion object {
+        private val log by logger(IntegrationValidator::class.java)
+    }
+
+    private val medeiaApi = MedeiaJacksonApi()
+
+    private fun loadSchema(): SchemaValidator {
+        val resource = javaClass.getResource("/schema/system/integration.schema")
+        resource ?: run {
+            log.error("failed to load integration schema resource")
+            throw IntegrationValidationException("failed to load integration schema resource")
+        }

```

Comment:
Thinking about extensibility, do you want this to take in a file and be able to validate it? Then this class can be re-used to validate other things like index pattern, etc.

Reply:
Good idea! Will definitely have to incorporate it.